src/main/java/com/fasterxml/jackson/databind/ext/KotlinSupport.java

```diff
@@ -0,0 +1,23 @@
+package com.fasterxml.jackson.databind.ext;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+
+public class KotlinSupport {
+    public static boolean isJvmInlineClassSyntheticConstructor(Constructor<?> ctor) {
+        Class<?>[] params = ctor.getParameterTypes();
+        if (params.length == 0) {
+            return false;
+        }
+
+        Class<?> lastParam = params[params.length - 1];
+        return ctor.isSynthetic() && lastParam.getName().equals("kotlin.jvm.internal.DefaultConstructorMarker");
+    }
+
+    public static boolean isJvmInlineClassSyntheticBoxingFunction(Method method) {
+        return Modifier.isStatic(method.getModifiers())
+                && method.isSynthetic()
+                && method.getName().equals("box-impl");
+    }
```

Comment:
As commented here, using only `box-impl` to deserialize `value class` may cause unexpected bugs.
https://github.com/FasterXML/jackson-databind/pull/4066#issuecomment-1665745100

Therefore, this change should be removed.